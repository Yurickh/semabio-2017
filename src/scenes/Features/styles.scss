@import 'ambience/responsive';
@import 'base/responsive-prop';

.page-features {
  @include responsive-prop(margin-top, 100, 170);

  & > .selector {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  & > .selector > .page-selector {
    @include responsive-prop(width, 90, 90);
    @include responsive-prop(height, 20, 20);
  }

  & > .selector-placeholder {
    @include responsive-prop(height, 20, 20);
  }

  & > .block-selector {
    text-align: center;

    @include responsive-prop(margin-top, 32, 32);
  }

  & > .content {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;

    width: 80vw;
    margin: 0 auto;

    @include responsive-prop(margin-top, 32, 32);
  }

  & > .content > * {
    width: 45%;
    margin-top: 1.5em;

    color: #000;

    @include responsive-prop(padding-right, 20, 20);
    @include responsive-prop(line-height, 20, 26);
    @include responsive-prop(font-size, 14, 18);
    @include small-only {
      width: 100%;
    }
  }
}
