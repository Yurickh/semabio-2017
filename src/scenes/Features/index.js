import React, { Component } from 'react'

import Title from '../../components/Title'
import ModeSelector from '../../components/ModeSelector'
import BlockSelector from '../../components/BlockSelector'

import featuresData from '../../helpers/featuresData'

import './styles.css'

class Features extends Component {
  data = featuresData()

  state = {
    currentMode: 0,
    currentBlock: 0,
  }

  changeMode = (mode) => {
    this.setState({ currentMode: mode })
  }

  changeBlock = (block) => {
    this.setState({ currentBlock: block })
  }

  fileNameFor = (index) => {
    const block = ['A', 'B', 'C', 'D', 'E'][this.state.currentBlock]
    const number = ('00' + index).slice(-2)
    const PUBLIC_URL = process.env.PUBLIC_URL

    return `${PUBLIC_URL}/ementas/${block}${number}.pdf`
  }

  renderContent = () => {
    switch(this.state.currentMode) {
      case 0: // Minicursos
        return this.data.minicourses[this.state.currentBlock]
          .map((miniCourse, index) => (
            <a href={this.fileNameFor(index+1)}
              target="_blank"
              key={miniCourse}
              rel="noopener noreferrer"
              className="minicourse"
              >
              {miniCourse}
            </a>
          ))
      case 1: // Palestras
        return this.data.lectures.map(lecture => {
          return <div
                  key={lecture} className="lecture">{lecture}
                </div>
        })
      case 2: // Rodas de conversa
        return this.data.talkingCircles.map(talkingCircle => {
          return <div
                  key={talkingCircle} className="circle">{talkingCircle}
                </div>
        })
      default: return null
    }
  }

  render() {
    return (
      <section className="page-features">
        <Title>Programação</Title>

        <ModeSelector
          currentMode={this.state.currentMode}
          onChange={this.changeMode}
        />

        {
          this.state.currentMode === 0 // Minicursos
          ? <BlockSelector
            currentBlock={this.state.currentBlock}
            onChange={this.changeBlock}
          />
          : null
        }

        <div className="content">
          { this.renderContent() }
        </div>

      </section>
    )
  }
}

export default Features
