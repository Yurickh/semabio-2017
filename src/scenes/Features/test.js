import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Features from '../Features'

describe('Features', () => {
  it('should render correctly', () => {
    const tree = shallow(<Features />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
