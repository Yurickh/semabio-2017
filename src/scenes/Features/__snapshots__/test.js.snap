// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features should render correctly 1`] = `
<_
  className="page-features"
  name="features"
>
  <Title>
    Programação
  </Title>
  <ModeSelector
    currentMode={0}
    onChange={[Function]}
  />
  <BlockSelector
    currentBlock={0}
    onChange={[Function]}
  />
  <div
    className="content"
  >
    <a
      className="minicourse"
      href="/menus/A01.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Invasões Biológicas
    </a>
    <a
      className="minicourse"
      href="/menus/A02.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Introdução aos Métodos de estudo em Anatomia Comparativa de Vertebrados
    </a>
    <a
      className="minicourse"
      href="/menus/A03.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Citogenética Animal
    </a>
    <a
      className="minicourse"
      href="/menus/A04.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Cultivo de células em Mamíferos
    </a>
    <a
      className="minicourse"
      href="/menus/A05.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Insetos e o homem: vamos discutir esta relação?
    </a>
    <a
      className="minicourse"
      href="/menus/A06.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Imunoterapias
    </a>
    <a
      className="minicourse"
      href="/menus/A07.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Produção de Biofármacos
    </a>
    <a
      className="minicourse"
      href="/menus/A08.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      O despertar dos sentidos: o que sentimos, por que sentimos e como sentimos?
    </a>
    <a
      className="minicourse"
      href="/menus/A09.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Oficina de Iniciação em Bonsai
    </a>
    <a
      className="minicourse"
      href="/menus/A10.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      Desenvolvimento e Aplicação de Nanopartículas Carreadoras de Fármacos
    </a>
  </div>
</_>
`;
